### DISCUSSION TITLE
app not updating with .apk file

### DISCUSSION THREAD

**Discussion by froghunter3** [KRTirtho/spotube#640] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

everytime i try to update the app using the .apk file it just says "app not installed" without any elaboration, also when is 3.1 coming to f-Droid?

### Expected Behavior

for the .apk file to update the app

### Steps to reproduce

i downloaded the .apk from github then clicked install then it just doesn't 

### Operating System

android

### Spotube version

updating 3.0.1 to 3.1.0

### Installation source

F-Droid (Android)

### Additional information

I changed the file songs download to, does that affect the update?

**Comment by KRTirtho** [comment]
This happens because F-Droid, PlayStore & Github apk signatures are different. It will only update if the signature match else will return error
So you've to uninstall the old version if you're updating the app from another source

And the release is usually delayed on F-Droid due to checks & custom builds. Same for PlayStore. It'll be available soon
Please do not create issues for this reason. We've discussion for this :smile:

**Comment by Nikoichu** [comment]
@froghunter3  There is actually a way to switch signature versions without losing your data. Use a backup app such as Neo Backup (on F-Droid) to back up the app data, then uninstall the current version, install the new one and restore the data with Neo Backup.
While it might produce some issues depending on the update and how it handles its data, it has worked flawlessly for me so far. It might require root however.
